Graphics/Slicer/Math/Skeleton/NodeTrees.hs

```diff
@@ -109,18 +111,18 @@ sortNodeTrees = sortBy compareNodeTrees
 
 -- dependent utility functions. used by internal components. not exported.
 
--- | Find a node with an output of the PLine given. Start at the most recent generation, and check backwards.
+-- | Find a node with an output of the PLine given. Check the most recent generation, and if recurse is set, check backwards.
 findINodeByOutput :: [[INode]] -> PLine2 -> Bool -> Maybe ([[INode]],INode)
 findINodeByOutput iNodeSets plineOut recurse
   | null iNodeSets = Nothing
   | otherwise = case nodesMatching of
                   [] -> if recurse
                         then case iNodeSets of
                                [] -> Nothing
-                               [_] -> Nothing
+                               [(INode _ _ _ _:_)] -> Nothing
                                (_:_) -> findINodeByOutput (init iNodeSets) plineOut recurse
                         else Nothing
-                  [_] -> Just (iNodeSets, head nodesMatching)
+                  [iNode] -> Just (iNodeSets, iNode)
                   (_:_) -> error "more than one node in a given generation with the same PLine out!"
```

Comment:
here it happens again: you should probably check all occurrances of `(_:_)` and double-check if you really meant "list of at least one element', or if you want to change it to `(_:_:_)`.